[2024-12-08T21:48:00.382+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2024-12-08T21:48:00.398+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: sparking_flow.python_job manual__2024-12-08T21:47:58.708477+00:00 [queued]>
[2024-12-08T21:48:00.404+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: sparking_flow.python_job manual__2024-12-08T21:47:58.708477+00:00 [queued]>
[2024-12-08T21:48:00.405+0000] {taskinstance.py:2866} INFO - Starting attempt 1 of 1
[2024-12-08T21:48:00.429+0000] {taskinstance.py:2889} INFO - Executing <Task(SparkSubmitOperator): python_job> on 2024-12-08 21:47:58.708477+00:00
[2024-12-08T21:48:00.440+0000] {logging_mixin.py:190} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:70 DeprecationWarning: This process (pid=185) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2024-12-08T21:48:00.441+0000] {standard_task_runner.py:72} INFO - Started process 190 to run task
[2024-12-08T21:48:00.439+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'sparking_flow', 'python_job', 'manual__2024-12-08T21:47:58.708477+00:00', '--job-id', '276', '--raw', '--subdir', 'DAGS_FOLDER/spark_***.py', '--cfg-path', '/tmp/tmp9ay3z1ob']
[2024-12-08T21:48:00.443+0000] {standard_task_runner.py:105} INFO - Job 276: Subtask python_job
[2024-12-08T21:48:00.491+0000] {task_command.py:467} INFO - Running <TaskInstance: sparking_flow.python_job manual__2024-12-08T21:47:58.708477+00:00 [running]> on host 2a892a3469ec
[2024-12-08T21:48:00.562+0000] {taskinstance.py:3132} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='Yusuf Ganiyu' AIRFLOW_CTX_DAG_ID='sparking_flow' AIRFLOW_CTX_TASK_ID='python_job' AIRFLOW_CTX_EXECUTION_DATE='2024-12-08T21:47:58.708477+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-12-08T21:47:58.708477+00:00'
[2024-12-08T21:48:00.563+0000] {taskinstance.py:731} INFO - ::endgroup::
[2024-12-08T21:48:00.583+0000] {base.py:84} INFO - Retrieving connection 'spark-conn'
[2024-12-08T21:48:00.584+0000] {spark_submit.py:473} INFO - Spark-Submit cmd: spark-submit --master spark://sparkingflow-spark-master-1:7077 --name arrow-spark --deploy-mode client jobs/python/wordcountjob.py
[2024-12-08T21:48:05.258+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:05 INFO SparkContext: Running Spark version 3.5.3
[2024-12-08T21:48:05.263+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:05 INFO SparkContext: OS info Linux, 6.10.14-linuxkit, aarch64
[2024-12-08T21:48:05.266+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:05 INFO SparkContext: Java version 17.0.13
[2024-12-08T21:48:05.457+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:05 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
[2024-12-08T21:48:05.774+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:05 INFO ResourceUtils: ==============================================================
[2024-12-08T21:48:05.776+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:05 INFO ResourceUtils: No custom resources configured for spark.driver.
[2024-12-08T21:48:05.779+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:05 INFO ResourceUtils: ==============================================================
[2024-12-08T21:48:05.783+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:05 INFO SparkContext: Submitted application: CSVReadExample
[2024-12-08T21:48:05.829+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:05 INFO ResourceProfile: Default ResourceProfile created, executor resources: Map(memory -> name: memory, amount: 1024, script: , vendor: , offHeap -> name: offHeap, amount: 0, script: , vendor: ), task resources: Map(cpus -> name: cpus, amount: 1.0)
[2024-12-08T21:48:05.843+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:05 INFO ResourceProfile: Limiting resource is cpu
[2024-12-08T21:48:05.849+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:05 INFO ResourceProfileManager: Added ResourceProfile id: 0
[2024-12-08T21:48:05.950+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:05 INFO SecurityManager: Changing view acls to: ***
[2024-12-08T21:48:05.951+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:05 INFO SecurityManager: Changing modify acls to: ***
[2024-12-08T21:48:05.958+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:05 INFO SecurityManager: Changing view acls groups to:
[2024-12-08T21:48:05.959+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:05 INFO SecurityManager: Changing modify acls groups to:
[2024-12-08T21:48:05.959+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:05 INFO SecurityManager: SecurityManager: authentication disabled; ui acls disabled; users with view permissions: ***; groups with view permissions: EMPTY; users with modify permissions: ***; groups with modify permissions: EMPTY
[2024-12-08T21:48:06.423+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:06 INFO Utils: Successfully started service 'sparkDriver' on port 35609.
[2024-12-08T21:48:06.464+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:06 INFO SparkEnv: Registering MapOutputTracker
[2024-12-08T21:48:06.532+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:06 INFO SparkEnv: Registering BlockManagerMaster
[2024-12-08T21:48:06.573+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:06 INFO BlockManagerMasterEndpoint: Using org.apache.spark.storage.DefaultTopologyMapper for getting topology information
[2024-12-08T21:48:06.576+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:06 INFO BlockManagerMasterEndpoint: BlockManagerMasterEndpoint up
[2024-12-08T21:48:06.596+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:06 INFO SparkEnv: Registering BlockManagerMasterHeartbeat
[2024-12-08T21:48:06.643+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:06 INFO DiskBlockManager: Created local directory at /tmp/blockmgr-42e33f8d-bb1a-4476-8ab8-d3d9044ffd1a
[2024-12-08T21:48:06.690+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:06 INFO MemoryStore: MemoryStore started with capacity 434.4 MiB
[2024-12-08T21:48:06.740+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:06 INFO SparkEnv: Registering OutputCommitCoordinator
[2024-12-08T21:48:06.971+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:06 INFO JettyUtils: Start Jetty 0.0.0.0:4040 for SparkUI
[2024-12-08T21:48:07.085+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:07 WARN Utils: Service 'SparkUI' could not bind on port 4040. Attempting port 4041.
[2024-12-08T21:48:07.086+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:07 WARN Utils: Service 'SparkUI' could not bind on port 4041. Attempting port 4042.
[2024-12-08T21:48:07.110+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:07 INFO Utils: Successfully started service 'SparkUI' on port 4042.
[2024-12-08T21:48:07.393+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:07 INFO StandaloneAppClient$ClientEndpoint: Connecting to master spark://sparkingflow-spark-master-1:7077...
[2024-12-08T21:48:07.473+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:07 INFO TransportClientFactory: Successfully created connection to sparkingflow-spark-master-1/172.18.0.5:7077 after 42 ms (0 ms spent in bootstraps)
[2024-12-08T21:48:07.608+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:07 INFO StandaloneSchedulerBackend: Connected to Spark cluster with app ID app-20241208214807-0001
[2024-12-08T21:48:07.619+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:07 INFO Utils: Successfully started service 'org.apache.spark.network.netty.NettyBlockTransferService' on port 42569.
[2024-12-08T21:48:07.620+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:07 INFO NettyBlockTransferService: Server created on 2a892a3469ec:42569
[2024-12-08T21:48:07.622+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:07 INFO BlockManager: Using org.apache.spark.storage.RandomBlockReplicationPolicy for block replication policy
[2024-12-08T21:48:07.629+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:07 INFO BlockManagerMaster: Registering BlockManager BlockManagerId(driver, 2a892a3469ec, 42569, None)
[2024-12-08T21:48:07.633+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:07 INFO BlockManagerMasterEndpoint: Registering block manager 2a892a3469ec:42569 with 434.4 MiB RAM, BlockManagerId(driver, 2a892a3469ec, 42569, None)
[2024-12-08T21:48:07.635+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:07 INFO BlockManagerMaster: Registered BlockManager BlockManagerId(driver, 2a892a3469ec, 42569, None)
[2024-12-08T21:48:07.638+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:07 INFO BlockManager: Initialized BlockManager: BlockManagerId(driver, 2a892a3469ec, 42569, None)
[2024-12-08T21:48:07.835+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:07 INFO StandaloneSchedulerBackend: SchedulerBackend is ready for scheduling beginning after reached minRegisteredResourcesRatio: 0.0
[2024-12-08T21:48:08.155+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:08 INFO SparkContext: Added file /tmp/tmp02qvgn72/test1234.csv at spark://2a892a3469ec:35609/files/test1234.csv with timestamp 1733694488154
[2024-12-08T21:48:08.157+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:08 INFO Utils: Copying /tmp/tmp02qvgn72/test1234.csv to /tmp/spark-599b3fc5-6013-4abc-acbf-ada9ecd2bebf/userFiles-987e30b1-407f-4269-83af-2674fc8dd64c/test1234.csv
[2024-12-08T21:48:08.189+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:08 INFO SharedState: Setting hive.metastore.warehouse.dir ('null') to the value of spark.sql.warehouse.dir.
[2024-12-08T21:48:08.191+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:08 INFO SharedState: Warehouse path is 'file:/opt/***/spark-warehouse'.
[2024-12-08T21:48:09.775+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:09 INFO InMemoryFileIndex: It took 87 ms to list leaf files for 1 paths.
[2024-12-08T21:48:09.876+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:09 INFO InMemoryFileIndex: It took 2 ms to list leaf files for 1 paths.
[2024-12-08T21:48:11.316+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:11 INFO StandaloneAppClient$ClientEndpoint: Executor added: app-20241208214807-0001/0 on worker-20241208214741-172.18.0.7-43841 (172.18.0.7:43841) with 2 core(s)
[2024-12-08T21:48:11.324+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:11 INFO StandaloneSchedulerBackend: Granted executor ID app-20241208214807-0001/0 on hostPort 172.18.0.7:43841 with 2 core(s), 1024.0 MiB RAM
[2024-12-08T21:48:11.412+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:11 INFO StandaloneAppClient$ClientEndpoint: Executor updated: app-20241208214807-0001/0 is now RUNNING
[2024-12-08T21:48:12.944+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:12 INFO FileSourceStrategy: Pushed Filters:
[2024-12-08T21:48:12.947+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:12 INFO FileSourceStrategy: Post-Scan Filters: (length(trim(value#0, None)) > 0)
[2024-12-08T21:48:13.578+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:13 INFO CodeGenerator: Code generated in 247.683375 ms
[2024-12-08T21:48:13.641+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:13 INFO MemoryStore: Block broadcast_0 stored as values in memory (estimated size 199.6 KiB, free 434.2 MiB)
[2024-12-08T21:48:13.711+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:13 INFO MemoryStore: Block broadcast_0_piece0 stored as bytes in memory (estimated size 34.4 KiB, free 434.2 MiB)
[2024-12-08T21:48:13.714+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:13 INFO BlockManagerInfo: Added broadcast_0_piece0 in memory on 2a892a3469ec:42569 (size: 34.4 KiB, free: 434.4 MiB)
[2024-12-08T21:48:13.720+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:13 INFO SparkContext: Created broadcast 0 from csv at NativeMethodAccessorImpl.java:0
[2024-12-08T21:48:13.737+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:13 INFO FileSourceScanExec: Planning scan with bin packing, max size: 4194304 bytes, open cost is considered as scanning 4194304 bytes.
[2024-12-08T21:48:13.843+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:13 INFO SparkContext: Starting job: csv at NativeMethodAccessorImpl.java:0
[2024-12-08T21:48:13.871+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:13 INFO DAGScheduler: Got job 0 (csv at NativeMethodAccessorImpl.java:0) with 1 output partitions
[2024-12-08T21:48:13.874+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:13 INFO DAGScheduler: Final stage: ResultStage 0 (csv at NativeMethodAccessorImpl.java:0)
[2024-12-08T21:48:13.876+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:13 INFO DAGScheduler: Parents of final stage: List()
[2024-12-08T21:48:13.877+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:13 INFO DAGScheduler: Missing parents: List()
[2024-12-08T21:48:13.878+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:13 INFO DAGScheduler: Submitting ResultStage 0 (MapPartitionsRDD[3] at csv at NativeMethodAccessorImpl.java:0), which has no missing parents
[2024-12-08T21:48:13.961+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:13 INFO MemoryStore: Block broadcast_1 stored as values in memory (estimated size 13.5 KiB, free 434.2 MiB)
[2024-12-08T21:48:13.973+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:13 INFO MemoryStore: Block broadcast_1_piece0 stored as bytes in memory (estimated size 6.4 KiB, free 434.2 MiB)
[2024-12-08T21:48:13.974+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:13 INFO BlockManagerInfo: Added broadcast_1_piece0 in memory on 2a892a3469ec:42569 (size: 6.4 KiB, free: 434.4 MiB)
[2024-12-08T21:48:13.988+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:13 INFO SparkContext: Created broadcast 1 from broadcast at DAGScheduler.scala:1585
[2024-12-08T21:48:14.015+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:14 INFO DAGScheduler: Submitting 1 missing tasks from ResultStage 0 (MapPartitionsRDD[3] at csv at NativeMethodAccessorImpl.java:0) (first 15 tasks are for partitions Vector(0))
[2024-12-08T21:48:14.017+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:14 INFO TaskSchedulerImpl: Adding task set 0.0 with 1 tasks resource profile 0
[2024-12-08T21:48:14.462+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:14 INFO StandaloneSchedulerBackend$StandaloneDriverEndpoint: Registered executor NettyRpcEndpointRef(spark-client://Executor) (172.18.0.7:33744) with ID 0,  ResourceProfileId 0
[2024-12-08T21:48:14.510+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:14 INFO BlockManagerMasterEndpoint: Registering block manager 172.18.0.7:36227 with 434.4 MiB RAM, BlockManagerId(0, 172.18.0.7, 36227, None)
[2024-12-08T21:48:14.557+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:14 INFO TaskSetManager: Starting task 0.0 in stage 0.0 (TID 0) (172.18.0.7, executor 0, partition 0, PROCESS_LOCAL, 9852 bytes)
[2024-12-08T21:48:14.779+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:14 INFO BlockManagerInfo: Added broadcast_1_piece0 in memory on 172.18.0.7:36227 (size: 6.4 KiB, free: 434.4 MiB)
[2024-12-08T21:48:15.361+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:15 INFO BlockManagerInfo: Added broadcast_0_piece0 in memory on 172.18.0.7:36227 (size: 34.4 KiB, free: 434.4 MiB)
[2024-12-08T21:48:15.708+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:15 WARN TaskSetManager: Lost task 0.0 in stage 0.0 (TID 0) (172.18.0.7 executor 0): org.apache.spark.SparkFileNotFoundException: File file:/tmp/spark-599b3fc5-6013-4abc-acbf-ada9ecd2bebf/userFiles-987e30b1-407f-4269-83af-2674fc8dd64c/test1234.csv does not exist
[2024-12-08T21:48:15.709+0000] {spark_submit.py:634} INFO - It is possible the underlying files have been updated. You can explicitly invalidate the cache in Spark by running 'REFRESH TABLE tableName' command in SQL or by recreating the Dataset/DataFrame involved.
[2024-12-08T21:48:15.709+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.errors.QueryExecutionErrors$.readCurrentFileNotFoundError(QueryExecutionErrors.scala:781)
[2024-12-08T21:48:15.710+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.org$apache$spark$sql$execution$datasources$FileScanRDD$$anon$$readCurrentFile(FileScanRDD.scala:220)
[2024-12-08T21:48:15.710+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.nextIterator(FileScanRDD.scala:279)
[2024-12-08T21:48:15.710+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.hasNext(FileScanRDD.scala:129)
[2024-12-08T21:48:15.710+0000] {spark_submit.py:634} INFO - at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2024-12-08T21:48:15.710+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.processNext(Unknown Source)
[2024-12-08T21:48:15.710+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
[2024-12-08T21:48:15.711+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.WholeStageCodegenEvaluatorFactory$WholeStageCodegenPartitionEvaluator$$anon$1.hasNext(WholeStageCodegenEvaluatorFactory.scala:43)
[2024-12-08T21:48:15.711+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.SparkPlan.$anonfun$getByteArrayRdd$1(SparkPlan.scala:388)
[2024-12-08T21:48:15.711+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.RDD.$anonfun$mapPartitionsInternal$2(RDD.scala:893)
[2024-12-08T21:48:15.712+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.RDD.$anonfun$mapPartitionsInternal$2$adapted(RDD.scala:893)
[2024-12-08T21:48:15.712+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
[2024-12-08T21:48:15.712+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:367)
[2024-12-08T21:48:15.713+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.RDD.iterator(RDD.scala:331)
[2024-12-08T21:48:15.713+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:93)
[2024-12-08T21:48:15.714+0000] {spark_submit.py:634} INFO - at org.apache.spark.TaskContext.runTaskWithListeners(TaskContext.scala:166)
[2024-12-08T21:48:15.714+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.Task.run(Task.scala:141)
[2024-12-08T21:48:15.714+0000] {spark_submit.py:634} INFO - at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$4(Executor.scala:620)
[2024-12-08T21:48:15.714+0000] {spark_submit.py:634} INFO - at org.apache.spark.util.SparkErrorUtils.tryWithSafeFinally(SparkErrorUtils.scala:64)
[2024-12-08T21:48:15.715+0000] {spark_submit.py:634} INFO - at org.apache.spark.util.SparkErrorUtils.tryWithSafeFinally$(SparkErrorUtils.scala:61)
[2024-12-08T21:48:15.715+0000] {spark_submit.py:634} INFO - at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:94)
[2024-12-08T21:48:15.715+0000] {spark_submit.py:634} INFO - at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:623)
[2024-12-08T21:48:15.715+0000] {spark_submit.py:634} INFO - at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
[2024-12-08T21:48:15.716+0000] {spark_submit.py:634} INFO - at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
[2024-12-08T21:48:15.716+0000] {spark_submit.py:634} INFO - at java.base/java.lang.Thread.run(Thread.java:840)
[2024-12-08T21:48:15.716+0000] {spark_submit.py:634} INFO - 
[2024-12-08T21:48:15.717+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:15 INFO TaskSetManager: Starting task 0.1 in stage 0.0 (TID 1) (172.18.0.7, executor 0, partition 0, PROCESS_LOCAL, 9852 bytes)
[2024-12-08T21:48:15.753+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:15 INFO TaskSetManager: Lost task 0.1 in stage 0.0 (TID 1) on 172.18.0.7, executor 0: org.apache.spark.SparkFileNotFoundException (File file:/tmp/spark-599b3fc5-6013-4abc-acbf-ada9ecd2bebf/userFiles-987e30b1-407f-4269-83af-2674fc8dd64c/test1234.csv does not exist
[2024-12-08T21:48:15.756+0000] {spark_submit.py:634} INFO - It is possible the underlying files have been updated. You can explicitly invalidate the cache in Spark by running 'REFRESH TABLE tableName' command in SQL or by recreating the Dataset/DataFrame involved.) [duplicate 1]
[2024-12-08T21:48:15.757+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:15 INFO TaskSetManager: Starting task 0.2 in stage 0.0 (TID 2) (172.18.0.7, executor 0, partition 0, PROCESS_LOCAL, 9852 bytes)
[2024-12-08T21:48:15.778+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:15 INFO TaskSetManager: Lost task 0.2 in stage 0.0 (TID 2) on 172.18.0.7, executor 0: org.apache.spark.SparkFileNotFoundException (File file:/tmp/spark-599b3fc5-6013-4abc-acbf-ada9ecd2bebf/userFiles-987e30b1-407f-4269-83af-2674fc8dd64c/test1234.csv does not exist
[2024-12-08T21:48:15.784+0000] {spark_submit.py:634} INFO - It is possible the underlying files have been updated. You can explicitly invalidate the cache in Spark by running 'REFRESH TABLE tableName' command in SQL or by recreating the Dataset/DataFrame involved.) [duplicate 2]
[2024-12-08T21:48:15.785+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:15 INFO TaskSetManager: Starting task 0.3 in stage 0.0 (TID 3) (172.18.0.7, executor 0, partition 0, PROCESS_LOCAL, 9852 bytes)
[2024-12-08T21:48:15.812+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:15 INFO TaskSetManager: Lost task 0.3 in stage 0.0 (TID 3) on 172.18.0.7, executor 0: org.apache.spark.SparkFileNotFoundException (File file:/tmp/spark-599b3fc5-6013-4abc-acbf-ada9ecd2bebf/userFiles-987e30b1-407f-4269-83af-2674fc8dd64c/test1234.csv does not exist
[2024-12-08T21:48:15.813+0000] {spark_submit.py:634} INFO - It is possible the underlying files have been updated. You can explicitly invalidate the cache in Spark by running 'REFRESH TABLE tableName' command in SQL or by recreating the Dataset/DataFrame involved.) [duplicate 3]
[2024-12-08T21:48:15.813+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:15 ERROR TaskSetManager: Task 0 in stage 0.0 failed 4 times; aborting job
[2024-12-08T21:48:15.814+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:15 INFO TaskSchedulerImpl: Removed TaskSet 0.0, whose tasks have all completed, from pool
[2024-12-08T21:48:15.818+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:15 INFO TaskSchedulerImpl: Cancelling stage 0
[2024-12-08T21:48:15.818+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:15 INFO TaskSchedulerImpl: Killing all running tasks in stage 0: Stage cancelled: Job aborted due to stage failure: Task 0 in stage 0.0 failed 4 times, most recent failure: Lost task 0.3 in stage 0.0 (TID 3) (172.18.0.7 executor 0): org.apache.spark.SparkFileNotFoundException: File file:/tmp/spark-599b3fc5-6013-4abc-acbf-ada9ecd2bebf/userFiles-987e30b1-407f-4269-83af-2674fc8dd64c/test1234.csv does not exist
[2024-12-08T21:48:15.819+0000] {spark_submit.py:634} INFO - It is possible the underlying files have been updated. You can explicitly invalidate the cache in Spark by running 'REFRESH TABLE tableName' command in SQL or by recreating the Dataset/DataFrame involved.
[2024-12-08T21:48:15.819+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.errors.QueryExecutionErrors$.readCurrentFileNotFoundError(QueryExecutionErrors.scala:781)
[2024-12-08T21:48:15.820+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.org$apache$spark$sql$execution$datasources$FileScanRDD$$anon$$readCurrentFile(FileScanRDD.scala:220)
[2024-12-08T21:48:15.820+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.nextIterator(FileScanRDD.scala:279)
[2024-12-08T21:48:15.821+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.hasNext(FileScanRDD.scala:129)
[2024-12-08T21:48:15.821+0000] {spark_submit.py:634} INFO - at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2024-12-08T21:48:15.821+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.processNext(Unknown Source)
[2024-12-08T21:48:15.822+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
[2024-12-08T21:48:15.824+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.WholeStageCodegenEvaluatorFactory$WholeStageCodegenPartitionEvaluator$$anon$1.hasNext(WholeStageCodegenEvaluatorFactory.scala:43)
[2024-12-08T21:48:15.825+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.SparkPlan.$anonfun$getByteArrayRdd$1(SparkPlan.scala:388)
[2024-12-08T21:48:15.826+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.RDD.$anonfun$mapPartitionsInternal$2(RDD.scala:893)
[2024-12-08T21:48:15.826+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.RDD.$anonfun$mapPartitionsInternal$2$adapted(RDD.scala:893)
[2024-12-08T21:48:15.827+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
[2024-12-08T21:48:15.827+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:367)
[2024-12-08T21:48:15.827+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.RDD.iterator(RDD.scala:331)
[2024-12-08T21:48:15.828+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:93)
[2024-12-08T21:48:15.828+0000] {spark_submit.py:634} INFO - at org.apache.spark.TaskContext.runTaskWithListeners(TaskContext.scala:166)
[2024-12-08T21:48:15.828+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.Task.run(Task.scala:141)
[2024-12-08T21:48:15.829+0000] {spark_submit.py:634} INFO - at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$4(Executor.scala:620)
[2024-12-08T21:48:15.829+0000] {spark_submit.py:634} INFO - at org.apache.spark.util.SparkErrorUtils.tryWithSafeFinally(SparkErrorUtils.scala:64)
[2024-12-08T21:48:15.829+0000] {spark_submit.py:634} INFO - at org.apache.spark.util.SparkErrorUtils.tryWithSafeFinally$(SparkErrorUtils.scala:61)
[2024-12-08T21:48:15.829+0000] {spark_submit.py:634} INFO - at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:94)
[2024-12-08T21:48:15.830+0000] {spark_submit.py:634} INFO - at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:623)
[2024-12-08T21:48:15.830+0000] {spark_submit.py:634} INFO - at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
[2024-12-08T21:48:15.831+0000] {spark_submit.py:634} INFO - at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
[2024-12-08T21:48:15.831+0000] {spark_submit.py:634} INFO - at java.base/java.lang.Thread.run(Thread.java:840)
[2024-12-08T21:48:15.831+0000] {spark_submit.py:634} INFO - 
[2024-12-08T21:48:15.831+0000] {spark_submit.py:634} INFO - Driver stacktrace:
[2024-12-08T21:48:15.831+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:15 INFO DAGScheduler: ResultStage 0 (csv at NativeMethodAccessorImpl.java:0) failed in 1.927 s due to Job aborted due to stage failure: Task 0 in stage 0.0 failed 4 times, most recent failure: Lost task 0.3 in stage 0.0 (TID 3) (172.18.0.7 executor 0): org.apache.spark.SparkFileNotFoundException: File file:/tmp/spark-599b3fc5-6013-4abc-acbf-ada9ecd2bebf/userFiles-987e30b1-407f-4269-83af-2674fc8dd64c/test1234.csv does not exist
[2024-12-08T21:48:15.831+0000] {spark_submit.py:634} INFO - It is possible the underlying files have been updated. You can explicitly invalidate the cache in Spark by running 'REFRESH TABLE tableName' command in SQL or by recreating the Dataset/DataFrame involved.
[2024-12-08T21:48:15.831+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.errors.QueryExecutionErrors$.readCurrentFileNotFoundError(QueryExecutionErrors.scala:781)
[2024-12-08T21:48:15.832+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.org$apache$spark$sql$execution$datasources$FileScanRDD$$anon$$readCurrentFile(FileScanRDD.scala:220)
[2024-12-08T21:48:15.832+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.nextIterator(FileScanRDD.scala:279)
[2024-12-08T21:48:15.832+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.hasNext(FileScanRDD.scala:129)
[2024-12-08T21:48:15.832+0000] {spark_submit.py:634} INFO - at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2024-12-08T21:48:15.833+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.processNext(Unknown Source)
[2024-12-08T21:48:15.833+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
[2024-12-08T21:48:15.834+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.WholeStageCodegenEvaluatorFactory$WholeStageCodegenPartitionEvaluator$$anon$1.hasNext(WholeStageCodegenEvaluatorFactory.scala:43)
[2024-12-08T21:48:15.834+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.SparkPlan.$anonfun$getByteArrayRdd$1(SparkPlan.scala:388)
[2024-12-08T21:48:15.834+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.RDD.$anonfun$mapPartitionsInternal$2(RDD.scala:893)
[2024-12-08T21:48:15.835+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.RDD.$anonfun$mapPartitionsInternal$2$adapted(RDD.scala:893)
[2024-12-08T21:48:15.835+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
[2024-12-08T21:48:15.836+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:367)
[2024-12-08T21:48:15.836+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.RDD.iterator(RDD.scala:331)
[2024-12-08T21:48:15.836+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:93)
[2024-12-08T21:48:15.837+0000] {spark_submit.py:634} INFO - at org.apache.spark.TaskContext.runTaskWithListeners(TaskContext.scala:166)
[2024-12-08T21:48:15.837+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.Task.run(Task.scala:141)
[2024-12-08T21:48:15.837+0000] {spark_submit.py:634} INFO - at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$4(Executor.scala:620)
[2024-12-08T21:48:15.838+0000] {spark_submit.py:634} INFO - at org.apache.spark.util.SparkErrorUtils.tryWithSafeFinally(SparkErrorUtils.scala:64)
[2024-12-08T21:48:15.838+0000] {spark_submit.py:634} INFO - at org.apache.spark.util.SparkErrorUtils.tryWithSafeFinally$(SparkErrorUtils.scala:61)
[2024-12-08T21:48:15.838+0000] {spark_submit.py:634} INFO - at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:94)
[2024-12-08T21:48:15.839+0000] {spark_submit.py:634} INFO - at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:623)
[2024-12-08T21:48:15.839+0000] {spark_submit.py:634} INFO - at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
[2024-12-08T21:48:15.839+0000] {spark_submit.py:634} INFO - at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
[2024-12-08T21:48:15.839+0000] {spark_submit.py:634} INFO - at java.base/java.lang.Thread.run(Thread.java:840)
[2024-12-08T21:48:15.839+0000] {spark_submit.py:634} INFO - 
[2024-12-08T21:48:15.840+0000] {spark_submit.py:634} INFO - Driver stacktrace:
[2024-12-08T21:48:15.840+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:15 INFO DAGScheduler: Job 0 failed: csv at NativeMethodAccessorImpl.java:0, took 1.984054 s
[2024-12-08T21:48:16.050+0000] {spark_submit.py:634} INFO - Traceback (most recent call last):
[2024-12-08T21:48:16.051+0000] {spark_submit.py:634} INFO - File "/opt/***/jobs/python/wordcountjob.py", line 32, in <module>
[2024-12-08T21:48:16.051+0000] {spark_submit.py:634} INFO - df = spark.read.csv(path=myfile, header=True, inferSchema=True)
[2024-12-08T21:48:16.052+0000] {spark_submit.py:634} INFO - ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2024-12-08T21:48:16.052+0000] {spark_submit.py:634} INFO - File "/home/***/.local/lib/python3.12/site-packages/pyspark/python/lib/pyspark.zip/pyspark/sql/readwriter.py", line 740, in csv
[2024-12-08T21:48:16.054+0000] {spark_submit.py:634} INFO - File "/home/***/.local/lib/python3.12/site-packages/pyspark/python/lib/py4j-0.10.9.7-src.zip/py4j/java_gateway.py", line 1322, in __call__
[2024-12-08T21:48:16.054+0000] {spark_submit.py:634} INFO - File "/home/***/.local/lib/python3.12/site-packages/pyspark/python/lib/pyspark.zip/pyspark/errors/exceptions/captured.py", line 179, in deco
[2024-12-08T21:48:16.054+0000] {spark_submit.py:634} INFO - File "/home/***/.local/lib/python3.12/site-packages/pyspark/python/lib/py4j-0.10.9.7-src.zip/py4j/protocol.py", line 326, in get_return_value
[2024-12-08T21:48:16.097+0000] {spark_submit.py:634} INFO - py4j.protocol.Py4JJavaError: An error occurred while calling o27.csv.
[2024-12-08T21:48:16.097+0000] {spark_submit.py:634} INFO - : org.apache.spark.SparkException: Job aborted due to stage failure: Task 0 in stage 0.0 failed 4 times, most recent failure: Lost task 0.3 in stage 0.0 (TID 3) (172.18.0.7 executor 0): org.apache.spark.SparkFileNotFoundException: File file:/tmp/spark-599b3fc5-6013-4abc-acbf-ada9ecd2bebf/userFiles-987e30b1-407f-4269-83af-2674fc8dd64c/test1234.csv does not exist
[2024-12-08T21:48:16.098+0000] {spark_submit.py:634} INFO - It is possible the underlying files have been updated. You can explicitly invalidate the cache in Spark by running 'REFRESH TABLE tableName' command in SQL or by recreating the Dataset/DataFrame involved.
[2024-12-08T21:48:16.098+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.errors.QueryExecutionErrors$.readCurrentFileNotFoundError(QueryExecutionErrors.scala:781)
[2024-12-08T21:48:16.098+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.org$apache$spark$sql$execution$datasources$FileScanRDD$$anon$$readCurrentFile(FileScanRDD.scala:220)
[2024-12-08T21:48:16.099+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.nextIterator(FileScanRDD.scala:279)
[2024-12-08T21:48:16.099+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.hasNext(FileScanRDD.scala:129)
[2024-12-08T21:48:16.099+0000] {spark_submit.py:634} INFO - at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2024-12-08T21:48:16.100+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.processNext(Unknown Source)
[2024-12-08T21:48:16.100+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
[2024-12-08T21:48:16.100+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.WholeStageCodegenEvaluatorFactory$WholeStageCodegenPartitionEvaluator$$anon$1.hasNext(WholeStageCodegenEvaluatorFactory.scala:43)
[2024-12-08T21:48:16.101+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.SparkPlan.$anonfun$getByteArrayRdd$1(SparkPlan.scala:388)
[2024-12-08T21:48:16.101+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.RDD.$anonfun$mapPartitionsInternal$2(RDD.scala:893)
[2024-12-08T21:48:16.101+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.RDD.$anonfun$mapPartitionsInternal$2$adapted(RDD.scala:893)
[2024-12-08T21:48:16.102+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
[2024-12-08T21:48:16.102+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:367)
[2024-12-08T21:48:16.102+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.RDD.iterator(RDD.scala:331)
[2024-12-08T21:48:16.103+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:93)
[2024-12-08T21:48:16.103+0000] {spark_submit.py:634} INFO - at org.apache.spark.TaskContext.runTaskWithListeners(TaskContext.scala:166)
[2024-12-08T21:48:16.103+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.Task.run(Task.scala:141)
[2024-12-08T21:48:16.103+0000] {spark_submit.py:634} INFO - at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$4(Executor.scala:620)
[2024-12-08T21:48:16.104+0000] {spark_submit.py:634} INFO - at org.apache.spark.util.SparkErrorUtils.tryWithSafeFinally(SparkErrorUtils.scala:64)
[2024-12-08T21:48:16.104+0000] {spark_submit.py:634} INFO - at org.apache.spark.util.SparkErrorUtils.tryWithSafeFinally$(SparkErrorUtils.scala:61)
[2024-12-08T21:48:16.104+0000] {spark_submit.py:634} INFO - at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:94)
[2024-12-08T21:48:16.105+0000] {spark_submit.py:634} INFO - at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:623)
[2024-12-08T21:48:16.105+0000] {spark_submit.py:634} INFO - at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
[2024-12-08T21:48:16.105+0000] {spark_submit.py:634} INFO - at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
[2024-12-08T21:48:16.105+0000] {spark_submit.py:634} INFO - at java.base/java.lang.Thread.run(Thread.java:840)
[2024-12-08T21:48:16.106+0000] {spark_submit.py:634} INFO - 
[2024-12-08T21:48:16.106+0000] {spark_submit.py:634} INFO - Driver stacktrace:
[2024-12-08T21:48:16.106+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.DAGScheduler.failJobAndIndependentStages(DAGScheduler.scala:2856)
[2024-12-08T21:48:16.106+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.DAGScheduler.$anonfun$abortStage$2(DAGScheduler.scala:2792)
[2024-12-08T21:48:16.107+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.DAGScheduler.$anonfun$abortStage$2$adapted(DAGScheduler.scala:2791)
[2024-12-08T21:48:16.107+0000] {spark_submit.py:634} INFO - at scala.collection.mutable.ResizableArray.foreach(ResizableArray.scala:62)
[2024-12-08T21:48:16.107+0000] {spark_submit.py:634} INFO - at scala.collection.mutable.ResizableArray.foreach$(ResizableArray.scala:55)
[2024-12-08T21:48:16.107+0000] {spark_submit.py:634} INFO - at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:49)
[2024-12-08T21:48:16.108+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.DAGScheduler.abortStage(DAGScheduler.scala:2791)
[2024-12-08T21:48:16.108+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.DAGScheduler.$anonfun$handleTaskSetFailed$1(DAGScheduler.scala:1247)
[2024-12-08T21:48:16.108+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.DAGScheduler.$anonfun$handleTaskSetFailed$1$adapted(DAGScheduler.scala:1247)
[2024-12-08T21:48:16.108+0000] {spark_submit.py:634} INFO - at scala.Option.foreach(Option.scala:407)
[2024-12-08T21:48:16.109+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.DAGScheduler.handleTaskSetFailed(DAGScheduler.scala:1247)
[2024-12-08T21:48:16.109+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.doOnReceive(DAGScheduler.scala:3060)
[2024-12-08T21:48:16.109+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:2994)
[2024-12-08T21:48:16.109+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:2983)
[2024-12-08T21:48:16.110+0000] {spark_submit.py:634} INFO - at org.apache.spark.util.EventLoop$$anon$1.run(EventLoop.scala:49)
[2024-12-08T21:48:16.110+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.DAGScheduler.runJob(DAGScheduler.scala:989)
[2024-12-08T21:48:16.110+0000] {spark_submit.py:634} INFO - at org.apache.spark.SparkContext.runJob(SparkContext.scala:2393)
[2024-12-08T21:48:16.110+0000] {spark_submit.py:634} INFO - at org.apache.spark.SparkContext.runJob(SparkContext.scala:2414)
[2024-12-08T21:48:16.111+0000] {spark_submit.py:634} INFO - at org.apache.spark.SparkContext.runJob(SparkContext.scala:2433)
[2024-12-08T21:48:16.111+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.SparkPlan.executeTake(SparkPlan.scala:530)
[2024-12-08T21:48:16.111+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.SparkPlan.executeTake(SparkPlan.scala:483)
[2024-12-08T21:48:16.112+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.CollectLimitExec.executeCollect(limit.scala:61)
[2024-12-08T21:48:16.112+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.Dataset.collectFromPlan(Dataset.scala:4333)
[2024-12-08T21:48:16.113+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.Dataset.$anonfun$head$1(Dataset.scala:3316)
[2024-12-08T21:48:16.114+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.Dataset.$anonfun$withAction$2(Dataset.scala:4323)
[2024-12-08T21:48:16.114+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.QueryExecution$.withInternalError(QueryExecution.scala:546)
[2024-12-08T21:48:16.114+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.Dataset.$anonfun$withAction$1(Dataset.scala:4321)
[2024-12-08T21:48:16.115+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId$6(SQLExecution.scala:125)
[2024-12-08T21:48:16.115+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.SQLExecution$.withSQLConfPropagated(SQLExecution.scala:201)
[2024-12-08T21:48:16.115+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId$1(SQLExecution.scala:108)
[2024-12-08T21:48:16.115+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:900)
[2024-12-08T21:48:16.115+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.SQLExecution$.withNewExecutionId(SQLExecution.scala:66)
[2024-12-08T21:48:16.115+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.Dataset.withAction(Dataset.scala:4321)
[2024-12-08T21:48:16.115+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.Dataset.head(Dataset.scala:3316)
[2024-12-08T21:48:16.116+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.Dataset.take(Dataset.scala:3539)
[2024-12-08T21:48:16.116+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.csv.TextInputCSVDataSource$.infer(CSVDataSource.scala:111)
[2024-12-08T21:48:16.116+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.csv.CSVDataSource.inferSchema(CSVDataSource.scala:64)
[2024-12-08T21:48:16.116+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.csv.CSVFileFormat.inferSchema(CSVFileFormat.scala:62)
[2024-12-08T21:48:16.116+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.DataSource.$anonfun$getOrInferFileFormatSchema$11(DataSource.scala:208)
[2024-12-08T21:48:16.116+0000] {spark_submit.py:634} INFO - at scala.Option.orElse(Option.scala:447)
[2024-12-08T21:48:16.116+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.DataSource.getOrInferFileFormatSchema(DataSource.scala:205)
[2024-12-08T21:48:16.117+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.DataSource.resolveRelation(DataSource.scala:407)
[2024-12-08T21:48:16.117+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.DataFrameReader.loadV1Source(DataFrameReader.scala:229)
[2024-12-08T21:48:16.117+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.DataFrameReader.$anonfun$load$2(DataFrameReader.scala:211)
[2024-12-08T21:48:16.118+0000] {spark_submit.py:634} INFO - at scala.Option.getOrElse(Option.scala:189)
[2024-12-08T21:48:16.118+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.DataFrameReader.load(DataFrameReader.scala:211)
[2024-12-08T21:48:16.118+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.DataFrameReader.csv(DataFrameReader.scala:538)
[2024-12-08T21:48:16.118+0000] {spark_submit.py:634} INFO - at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[2024-12-08T21:48:16.119+0000] {spark_submit.py:634} INFO - at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
[2024-12-08T21:48:16.119+0000] {spark_submit.py:634} INFO - at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[2024-12-08T21:48:16.120+0000] {spark_submit.py:634} INFO - at java.base/java.lang.reflect.Method.invoke(Method.java:569)
[2024-12-08T21:48:16.120+0000] {spark_submit.py:634} INFO - at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
[2024-12-08T21:48:16.121+0000] {spark_submit.py:634} INFO - at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:374)
[2024-12-08T21:48:16.121+0000] {spark_submit.py:634} INFO - at py4j.Gateway.invoke(Gateway.java:282)
[2024-12-08T21:48:16.122+0000] {spark_submit.py:634} INFO - at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
[2024-12-08T21:48:16.122+0000] {spark_submit.py:634} INFO - at py4j.commands.CallCommand.execute(CallCommand.java:79)
[2024-12-08T21:48:16.122+0000] {spark_submit.py:634} INFO - at py4j.ClientServerConnection.waitForCommands(ClientServerConnection.java:182)
[2024-12-08T21:48:16.123+0000] {spark_submit.py:634} INFO - at py4j.ClientServerConnection.run(ClientServerConnection.java:106)
[2024-12-08T21:48:16.123+0000] {spark_submit.py:634} INFO - at java.base/java.lang.Thread.run(Thread.java:840)
[2024-12-08T21:48:16.124+0000] {spark_submit.py:634} INFO - Caused by: org.apache.spark.SparkFileNotFoundException: File file:/tmp/spark-599b3fc5-6013-4abc-acbf-ada9ecd2bebf/userFiles-987e30b1-407f-4269-83af-2674fc8dd64c/test1234.csv does not exist
[2024-12-08T21:48:16.124+0000] {spark_submit.py:634} INFO - It is possible the underlying files have been updated. You can explicitly invalidate the cache in Spark by running 'REFRESH TABLE tableName' command in SQL or by recreating the Dataset/DataFrame involved.
[2024-12-08T21:48:16.124+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.errors.QueryExecutionErrors$.readCurrentFileNotFoundError(QueryExecutionErrors.scala:781)
[2024-12-08T21:48:16.124+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.org$apache$spark$sql$execution$datasources$FileScanRDD$$anon$$readCurrentFile(FileScanRDD.scala:220)
[2024-12-08T21:48:16.125+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.nextIterator(FileScanRDD.scala:279)
[2024-12-08T21:48:16.125+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.hasNext(FileScanRDD.scala:129)
[2024-12-08T21:48:16.126+0000] {spark_submit.py:634} INFO - at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2024-12-08T21:48:16.126+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.processNext(Unknown Source)
[2024-12-08T21:48:16.127+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
[2024-12-08T21:48:16.127+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.WholeStageCodegenEvaluatorFactory$WholeStageCodegenPartitionEvaluator$$anon$1.hasNext(WholeStageCodegenEvaluatorFactory.scala:43)
[2024-12-08T21:48:16.127+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.SparkPlan.$anonfun$getByteArrayRdd$1(SparkPlan.scala:388)
[2024-12-08T21:48:16.128+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.RDD.$anonfun$mapPartitionsInternal$2(RDD.scala:893)
[2024-12-08T21:48:16.128+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.RDD.$anonfun$mapPartitionsInternal$2$adapted(RDD.scala:893)
[2024-12-08T21:48:16.128+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
[2024-12-08T21:48:16.128+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:367)
[2024-12-08T21:48:16.129+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.RDD.iterator(RDD.scala:331)
[2024-12-08T21:48:16.129+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:93)
[2024-12-08T21:48:16.129+0000] {spark_submit.py:634} INFO - at org.apache.spark.TaskContext.runTaskWithListeners(TaskContext.scala:166)
[2024-12-08T21:48:16.129+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.Task.run(Task.scala:141)
[2024-12-08T21:48:16.129+0000] {spark_submit.py:634} INFO - at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$4(Executor.scala:620)
[2024-12-08T21:48:16.130+0000] {spark_submit.py:634} INFO - at org.apache.spark.util.SparkErrorUtils.tryWithSafeFinally(SparkErrorUtils.scala:64)
[2024-12-08T21:48:16.130+0000] {spark_submit.py:634} INFO - at org.apache.spark.util.SparkErrorUtils.tryWithSafeFinally$(SparkErrorUtils.scala:61)
[2024-12-08T21:48:16.130+0000] {spark_submit.py:634} INFO - at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:94)
[2024-12-08T21:48:16.130+0000] {spark_submit.py:634} INFO - at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:623)
[2024-12-08T21:48:16.130+0000] {spark_submit.py:634} INFO - at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
[2024-12-08T21:48:16.131+0000] {spark_submit.py:634} INFO - at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
[2024-12-08T21:48:16.131+0000] {spark_submit.py:634} INFO - ... 1 more
[2024-12-08T21:48:16.131+0000] {spark_submit.py:634} INFO - 
[2024-12-08T21:48:16.190+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:16 INFO SparkContext: Invoking stop() from shutdown hook
[2024-12-08T21:48:16.190+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:16 INFO SparkContext: SparkContext is stopping with exitCode 0.
[2024-12-08T21:48:16.201+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:16 INFO SparkUI: Stopped Spark web UI at http://2a892a3469ec:4042
[2024-12-08T21:48:16.208+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:16 INFO StandaloneSchedulerBackend: Shutting down all executors
[2024-12-08T21:48:16.208+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:16 INFO StandaloneSchedulerBackend$StandaloneDriverEndpoint: Asking each executor to shut down
[2024-12-08T21:48:16.230+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:16 INFO MapOutputTrackerMasterEndpoint: MapOutputTrackerMasterEndpoint stopped!
[2024-12-08T21:48:16.262+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:16 INFO MemoryStore: MemoryStore cleared
[2024-12-08T21:48:16.262+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:16 INFO BlockManager: BlockManager stopped
[2024-12-08T21:48:16.268+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:16 INFO BlockManagerMaster: BlockManagerMaster stopped
[2024-12-08T21:48:16.270+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:16 INFO OutputCommitCoordinator$OutputCommitCoordinatorEndpoint: OutputCommitCoordinator stopped!
[2024-12-08T21:48:16.285+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:16 INFO SparkContext: Successfully stopped SparkContext
[2024-12-08T21:48:16.286+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:16 INFO ShutdownHookManager: Shutdown hook called
[2024-12-08T21:48:16.286+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:16 INFO ShutdownHookManager: Deleting directory /tmp/spark-599b3fc5-6013-4abc-acbf-ada9ecd2bebf/pyspark-95fd7400-0129-4523-9177-f80ffc288c95
[2024-12-08T21:48:16.290+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:16 INFO ShutdownHookManager: Deleting directory /tmp/spark-ee604df3-abb9-4793-be60-b7d559798963
[2024-12-08T21:48:16.295+0000] {spark_submit.py:634} INFO - 24/12/08 21:48:16 INFO ShutdownHookManager: Deleting directory /tmp/spark-599b3fc5-6013-4abc-acbf-ada9ecd2bebf
[2024-12-08T21:48:16.350+0000] {taskinstance.py:3311} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 417, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 176, in execute
    self._hook.submit(self.application)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 560, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://sparkingflow-spark-master-1:7077 --name arrow-spark --deploy-mode client jobs/python/wordcountjob.py. Error code is: 1.
[2024-12-08T21:48:16.360+0000] {taskinstance.py:1225} INFO - Marking task as FAILED. dag_id=sparking_flow, task_id=python_job, run_id=manual__2024-12-08T21:47:58.708477+00:00, execution_date=20241208T214758, start_date=20241208T214800, end_date=20241208T214816
[2024-12-08T21:48:16.374+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2024-12-08T21:48:16.374+0000] {standard_task_runner.py:124} ERROR - Failed to execute job 276 for task python_job (Cannot execute: spark-submit --master spark://sparkingflow-spark-master-1:7077 --name arrow-spark --deploy-mode client jobs/python/wordcountjob.py. Error code is: 1.; 190)
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/cli.py", line 116, in wrapper
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
           ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3005, in _run_raw_task
    return _run_raw_task(
           ^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 273, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3159, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3183, in _execute_task
    return _execute_task(self, context, task_orig)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 417, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 176, in execute
    self._hook.submit(self.application)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 560, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://sparkingflow-spark-master-1:7077 --name arrow-spark --deploy-mode client jobs/python/wordcountjob.py. Error code is: 1.
[2024-12-08T21:48:16.398+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 1
[2024-12-08T21:48:16.419+0000] {taskinstance.py:3895} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-12-08T21:48:16.422+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
